pkg/providers/instancetype/instancetype.go

```diff
@@ -252,6 +264,58 @@ func (p *DefaultProvider) UpdateInstanceTypeOfferings(ctx context.Context) error
 	return nil
 }
 
+func (p *DefaultProvider) UpdateInstanceTypeMemoryOverhead(ctx context.Context, kubeClient client.Client) error {
+	nodeClassList := &v1.EC2NodeClassList{}
+	if err := kubeClient.List(ctx, nodeClassList); err != nil {
+		return fmt.Errorf("failed to list nodeclasses: %w", err)
+	}
+
+	nodeClassMap := lo.Associate(nodeClassList.Items, func(nodeClass v1.EC2NodeClass) (string, v1.EC2NodeClass) {
+		return nodeClass.Hash(), nodeClass
```

Comment:
Not sure you want to do the hash, as that encompasses more details than we care about for resource allocation.

Reply:
Further down there's a match against this map and the node's nodeclass hash annotation so that we're only storing cache values for the latest version of the nodeclass. Would you instead prefer keying this map on the nodeclass name and then while looping through the nodes having a check that the node's ami is within the nodeclass status.amis list?

Reply:
Updated with a lookup of AMI info instead